scripts/generate_iam_role.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+# Helper script to generate an IAM Role needed to install operator using role-based authentication.
+# https://sagemaker.readthedocs.io/en/stable/amazon_sagemaker_operators_for_kubernetes.html#create-an-iam-role
+#
+# Run as:
+# $ ./generate_iam_role.sh ${cluster_arn} ${operator_namespace} ${role_name}
+#
+
+CLUSTER_ARN=${1}
+OPERATOR_NAMESPACE=${2}
+ROLE_NAME=${3}
+aws_account=$(aws sts get-caller-identity --query Account --output text)
+trustfile="trust.json"
+
+# Use the cluster arn to get the region and cluster name
+# example, cluster_arn=arn:aws:eks:us-east-1:12345678910:cluster/test
+cluster_name=$(echo ${CLUSTER_ARN} | cut -d'/' -f2)
+cluster_region=$(echo ${CLUSTER_ARN} | cut -d':' -f4)
+
+# A function to get the OIDC_ID associated with an EKS cluster
+function get_oidc_id {
+    # TODO: Ideally this should be based on version compatibility instead of command failure
+    eksctl utils associate-iam-oidc-provider --cluster ${cluster_name} --region ${cluster_region} --approve
+    if [[ $? -ge 1 ]]; then
+        eksctl utils associate-iam-oidc-provider --name ${cluster_name} --region ${cluster_region} --approve
```

Comment:
Is this the issue with eksctl that the change the parameter name ?

Reply:
We don't really need to support the older `eksctl` versions since the images in which we run integration tests pull the latest versions.

Reply:
but it might fail in someone's local setup, like mine :)

Reply:
I thought you were running this from inside the integration test container? That has a reproducible environment that we would recommend contributors to use.

Reply:
but the script itself can be used standalone as well right.

Reply:
It was never developed with that as a consideration